src/test/java/org/broadinstitute/hellbender/utils/SATagBuilderUnitTests.java

```diff
@@ -0,0 +1,183 @@
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SATagBuilderUnitTests {
+
+    @Test
+    public static void testSetReadsAsSupplementalNo() {
+        final SAMFileHeader header = ArtificialReadUtils.createArtificialSamHeader();
+        GATKRead primarySup = ArtificialReadUtils.createArtificialRead(header, "read1", 1, 10000, new byte[] {}, new byte[] {}, "4M");
+        primarySup.setMappingQuality(100);
+        primarySup.setAttribute("NM",20);
+        primarySup.setIsReverseStrand(true);
+        GATKRead secondarySup = ArtificialReadUtils.createArtificialRead(header, "read2", 2, 10001, new byte[] {}, new byte[] {}, "4S");
+        secondarySup.setMappingQuality(200);
+        List<GATKRead> sups = new ArrayList<>();
+        sups.add(secondarySup);
+
+        SATagBuilder.setReadsAsSupplemental(primarySup,sups);
+        Assert.assertEquals(primarySup.getAttributeAsString("SA"), "3,10001,+,4S,200,*;");
+        Assert.assertEquals(primarySup.isSupplementaryAlignment(), false);
+        Assert.assertEquals(secondarySup.getAttributeAsString("SA"), "2,10000,-,4M,100,20;");
+        Assert.assertEquals(secondarySup.isSupplementaryAlignment(), true);
+
+        GATKRead tertiarySup = ArtificialReadUtils.createArtificialRead(header, "read3", 3, 10003, new byte[] {}, new byte[] {}, "4D");
+        tertiarySup.setMappingQuality(200);
+        primarySup.clearAttribute("SA");
+        secondarySup.clearAttribute("SA");
+        sups.add(tertiarySup);
+
+        SATagBuilder.setReadsAsSupplemental(primarySup,sups);
+        Assert.assertEquals(sups.size(), 2);
+        Assert.assertEquals(sups.get(0).getAttributeAsString("SA"), "2,10000,-,4M,100,20;4,10003,+,4D,200,*;");
+        Assert.assertTrue(sups.get(1).getAttributeAsString("SA").startsWith("2,10000,-,4M,100,20;"));
+        Assert.assertTrue(primarySup.getAttributeAsString("SA").contains("4,10003,+,4D,200,*;"));
+        Assert.assertTrue(primarySup.getAttributeAsString("SA").contains("3,10001,+,4S,200,*;"));
+
+        GATKRead unmappedSup = ArtificialReadUtils.createArtificialUnmappedRead(header,new byte[] {}, new byte[] {});
+        primarySup.clearAttribute("SA");
+        sups.clear();
+        sups.add(unmappedSup);
+
+        SATagBuilder.setReadsAsSupplemental(primarySup,sups);
+        Assert.assertEquals(primarySup.getAttributeAsString("SA"), "*,0,+,*,0,*;");
+        Assert.assertEquals(sups.get(0).getAttributeAsString("SA"), "2,10000,-,4M,100,20;");
+    }
+
+    @Test
+    public static void testRemoveTagFunctionality() {
+        final SAMFileHeader header = ArtificialReadUtils.createArtificialSamHeader();
+        GATKRead testRead = ArtificialReadUtils.createArtificialRead(header, "read1", 1, 10002, new byte[] {}, new byte[] {}, "4M");
+        testRead.setAttribute("SA","2,10000,+,4M,100,20;4,10003,+,4D,200,0;2,10000,-,4M,100,20");
```

Comment:
Can you extract this string to make it clear that it doesn't change between this line and line 71?